ribs/optimizers/_optimizer.py

```diff
@@ -93,6 +95,39 @@ def emitters(self):
         in this optimizer."""
         return self._emitters
 
+    def ask_dqd(self):
+        """Generates a batch of solutions by calling ask_dqd() on all DQD
+        emitters.
+
+        .. note:: The order of the solutions returned from this method is
+            important, so do not rearrange them.
+
+        Returns:
+            (n_solutions, dim) array: An array of n solutions to evaluate. Each
+            row contains a single solution.
+        Raises:
+            RuntimeError: This method was called without first calling
+                :meth:`tell`.
+        """
+        if self._last_called in ["ask", "ask_dqd"]:
+            raise RuntimeError("ask_dqd cannot be called immediately after " +
+                               self._last_called)
+        self._last_called = "ask_dqd"
+
+        self._solution_batch = []
+
+        for i, emitter in enumerate(self._emitters):
+            if isinstance(emitter, DQDEmitterBase):
+                emitter_sols = emitter.ask_dqd()
+                self._solution_batch.append(emitter_sols)
+                self._num_emitted[i] = len(emitter_sols)
+
+        # In case the emitters didn't return any solutions
+        self._solution_batch = np.concatenate(
+            self._solution_batch, axis=0) if self._solution_batch else np.array(
```

Comment:
`np.array((0, self._solution_dim))` returns an array with entries `0` and `self._solution_dim`. I think you should use `np.empty` here. Can you also make a test which verifies that we get the correct behavior?

Reply:
You're right. Just tested it in Python terminal. I'll make a test case for this